src/Psalm/Internal/Analyzer/Statements/Expression/BinaryOp/NonDivArithmeticOpAnalyzer.php

```diff
@@ -273,6 +275,19 @@ public static function analyze(
         }
     }
 
+    /**
+     * @param int|float $result
+     */
+    private static function getNumericalType($result): Type\Union
+    {
+        if ($result <= PHP_INT_MAX && $result >= PHP_INT_MIN) {
+            assert(is_int($result));
+            return Type::getInt(false, $result);
+        }
```

Comment:
```suggestion
        if (is_int($result)) {
            return Type::getInt(false, $result);
        }
```
I think it's the most robust way to check that.

Reply:
Yep, I should have gone there from the start!

Reply:
Yes thanks !